panel/src/api/request.js

```diff
@@ -1,3 +1,18 @@
+
+export async function toJson(response) {
+  const text = await response.text();
+  let data;
+
+  try {
+    data = JSON.parse(text);
+  } catch (e) {
+    window.panel.$vue.$store.dispatch("fatal", text);
+    throw new Error("The JSON response of the API could not be parsed");
```

Comment:
This duplicates `Vue.$api.onParserError()`. Is there any reason why we cannot reuse this method? This would reduce the risk of later regressions when we forget to change the code in one of the two places.

Reply:
Oh true, I didn't realise that with my change (basically combining the same code from API and Fiber in this `toJSON` method), I rendered `Vue.$api.onParserError()` as dead code. It never gets called actually anymore. We could either remove it or call it from `toJSON` - a bit weird cause then it gets called for FIber request failures as well, but it wouldn't really matter.

Reply:
TBH I don‘t fully understand how it all plays together. So I‘m fine with whatever makes the most sense for you.